File: 2128-ReversePrefixOfWord/2128-ReversePrefixOfWord.js
// Last updated: 10/22/2025, 12:08:54 AM
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function(word, ch) {
    let wordArray = word.split("")
    let left = 0
    let right = wordArray.indexOf(ch)

    while(left < right){
        [wordArray[left],wordArray[right]] = [wordArray[right],wordArray[left]]
        left++
        right--
    }

    return wordArray.join("")
    
};

File: 2128-ReversePrefixOfWord/2128-ReversePrefixOfWord.py
# Last updated: 10/22/2025, 12:08:57 AM
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        l=0
        r=word.find(ch)
        word=list(word)
        while l < r:
            word[l],word[r]=word[r],word[l]
            l += 1
            r -= 1
        return "".join(word)



        
        